### PR TITLE
EXPERIMENTAL: Investigating poor test execution performance

### PR DESCRIPTION
**Pull request by josecoll**
Recent upgrades to JUnit 5 and Gradle 5.4.1 have regressed general test execution performance.

### CODE REVIEW

**Comment by chrisr3** [comment]
Or maybe try this?
```gradle
ext {
    quasar_exclusions = [
        "co.paralleluniverse.**",
        'com.esotericsoftware.**',
        'jdk**',
        'junit**',
        'kotlin**',
        'net.rubygrapefruit.**',
        'org.gradle.**',
        'org.apache.**',
        'org.jacoco.**',
        'org.junit**',
        'org.slf4j**',
        'worker.org.gradle.**'
    ]
}
```
with this in `:node`?
```gradle
quasar {
    excludePackages.addAll(
        "antlr**",
        "com.fasterxml.**",
        "com.github.benmanes.caffeine.**",
        "com.google.**",
        "com.lmax.**",
        "com.nhaarman.mockito_kotlin**",
        "com.zaxxer.**",
        "net.bytebuddy**",
        "io.github.classgraph**",
        "liquibase**",
        "net.i2p.crypto.**",
        "nonapi.io.github.classgraph.**",
        "org.apiguardian.**",
        "org.bouncycastle**",
        "org.codehaus.**",
        "org.h2**",
        "org.hamcrest**",
        "org.hibernate**",
        "org.jboss.**",
        "org.mockito**",
        "org.objenesis**",
        "org.opentest4j**",
        "org.w3c.**",
        "org.xml**", 
        "org.yaml**",
        "rx.**"
    ]
}
```

**Comment by chrisr3** [comment]
You can also use
```gradle
quasar {
    verbose = true
    ...
}
```
to see which classes Quasar is actually transforming.

**Comment by josecoll** [comment]
> Or maybe try this?
> 
> ```groovy-gradle
> ext {
>     quasar_exclusions = [
>         "co.paralleluniverse.**",
>         'com.esotericsoftware.**',
>         'jdk**',
>         'junit**',
>         'kotlin**',
>         'net.rubygrapefruit.**',
>         'org.gradle.**',
>         'org.apache.**',
>         'org.jacoco.**',
>         'org.junit**',
>         'org.slf4j**',
>         'worker.org.gradle.**'
>     ]
> }
> ```
> 
> with this in `:node`?
> 
> ```groovy-gradle
> quasar {
>     excludePackages.addAll(
>         "antlr**",
>         "com.codahale**",
>         "com.fasterxml.**",
>         "com.github.benmanes.caffeine.**",
>         "com.google.**",
>         "com.lmax.**",
>         "com.nhaarman.mockito_kotlin**",
>         "com.zaxxer.**",
>         "net.bytebuddy**",
>         "io.github.classgraph**",
>         "io.netty*",
>         "liquibase**",
>         "net.i2p.crypto.**",
>         "nonapi.io.github.classgraph.**",
>         "org.apiguardian.**",
>         "org.bouncycastle**",
>         "org.codehaus.**",
>         "org.h2**",
>         "org.hamcrest**",
>         "org.hibernate**",
>         "org.jboss.**",
>         "org.mockito**",
>         "org.objenesis**",
>         "org.opentest4j**",
>         "org.w3c.**",
>         "org.xml**", 
>         "org.yaml**",
>         "rx**"
>     ]
> }
> ```

Is this the same exclusion pattern as in Node capsule/DriverDSLImpl?

**Comment by chrisr3** [comment]
> Is this the same exclusion pattern as in Node capsule/DriverDSLImpl?
The lists will obviously be similar, except that I don't see things like Gradle or JaCoCo in those.

File: build.gradle [hunk 1]
```diff
@@ -163,6 +163,26 @@ allprojects {
         )
     }
 
+    // quasar exclusions upon agent code instrumentation at run-time
+    ext {
+        verbose = true
```
Comment (chrisr3) [inline build.gradle]: For even greater performance, perhaps try with `verbose = false`?

Comment (chrisr3) [inline build.gradle]: This `verbose = false` doesn't do anything - it must be a copy-and-paste error.

**Comment by josecoll** [comment]
Superseded by
https://github.com/corda/corda/pull/5161 and
https://github.com/corda/corda/pull/5173

**Pull request closed without merging**